### PR TITLE
Fix `image_threshold` bug with integer arrays on some systems

### PR DESCRIPTION
**Pull request by user27182**
Fix issue with `image_threshold` unexpectedly returning all zeros for integer scalar arrays. This seems to be build-specific. I am not able to reproduce this locally but the bug shows up in the docs.

See images of the bug here https://github.com/pyvista/pyvista/pull/6071#issuecomment-2118468493

### CODE REVIEW

**Comment by user27182** [comment]
From the failed tests it looks like it's a Linux and/or Python < 3.12 issue. The tests fail for some cases with integer inputs.

File: pyvista/core/filters/image_data.py [hunk 1]
```diff
@@ -407,13 +408,28 @@ def image_threshold(
...
-        alg = _vtk.vtkImageThreshold()
-        alg.SetInputDataObject(self)
         if scalars is None:
             set_default_active_scalars(self)
             field, scalars = self.active_scalars_info
         else:
             field = self.get_array_association(scalars, preference=preference)
+
+        # For some systems and/or versions of numpy, integer scalars won't threshold
+        # correctly. Cast to float in these cases.
```
Comment (tkoyama010) [inline pyvista/core/filters/image_data.py:418]:
```suggestion
        # For some systems and Numpy < 2.0, integer scalars won't threshold
        # correctly. Cast to float in these cases.
```

File: pyvista/core/filters/image_data.py [hunk 2]
```diff
@@ -407,13 +408,28 @@ def image_threshold(
...
+            array_dtype := self.active_scalars.dtype,
+            int,
+        ) and array_dtype != np.dtype(np.uint8)
+        cast_dtype = (
+            has_int_dtype
+            and int(np.__version__.split('.')[0]) < 2
+            and platform.system() in ['Darwin', 'Linux']
+        )
+        if cast_dtype:
+            self[scalars] = self[scalars].astype(float, casting='safe')
```
Comment (tkoyama010) [inline pyvista/core/filters/image_data.py:429]:
```suggestion
        self[scalars] = self[scalars].astype(float, casting='safe') if cast_dtype
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: pyvista/core/filters/image_data.py (hunk 2 above)
Comment (tkoyama010) [reply pyvista/core/filters/image_data.py:429]: This doesn't work closing.

**Review by tkoyama010 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**